ogr/ogrspatialreference.cpp

```diff
@@ -172,8 +178,34 @@ struct OGRSpatialReference::Private
     const char *nullifyTargetKeyIfPossible(const char *pszTargetKey);
 
     void refreshAxisMapping();
+
```

Comment:
Added this comment:
```
    // This structures enables locking during calls to OGRSpatialReference
    // public methods. Locking is only needed for instances of
    // OGRSpatialReference that have been asked to be thread-safe at
    // construction.
    // The lock is not just for a single call to OGRSpatialReference::Private,
    // but for the series of calls done by a OGRSpatialReference method.
    // We need a recursive mutex, because some OGRSpatialReference methods
    // may call other ones.
```